### PR TITLE
Undefined behavior

### PR DESCRIPTION
**Pull request by SrsBusiness**
1. Added -g flag for clang/gcc in Makefile
2. Removed casts on void pointers that made (erroneous) assumptions on alignment
3. 128-bit integers (UUIDS) are now treated as unused data until we decide they are important. This means we don't need to rely on the GNU type: __int128_t
4. reentrant_memmove() is no longer used in favor of memmove/memcpy because we no longer have asynchronous thread signals. Those functions definitions can be removed if we want.
5. Updated tests for format_packet() and decode_packet()

### CODE REVIEW

File: test/hexdump.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#include <stdio.h>
+
+void hexdump (char *desc, void *addr, int len)
```
Comment (JDongian) [inline test/hexdump.c]: did you write this code?

**Comment by JDongian** [comment]
1. @amlweems and I decided against this. Write a `make debug` if you want this behaviour.
2. good
3. Handling UUIDs is required for some player interactions. You should make your own int128 type if you want to replace GNU.
4. good. Delete the function definition as it will exist in the git history.
5. good

File: Makefile [hunk 2]
```diff
@@ -2,7 +2,7 @@ CC      = clang
 SA      = scan-build
 TARGET  = mcc
 
-CFLAGS  = -Wall -Isrc --std=gnu99
+CFLAGS  = -Wall -Isrc --std=gnu99 -g
```
Comment (JDongian) [inline Makefile]: no

File: test/hexdump.c (hunk 1 above)
Comment (SrsBusiness) [inline test/hexdump.c]: Those files are copy-pasta from stack overflow

**Comment by SrsBusiness** [comment]
How about the -g flag for test?

**Comment by SrsBusiness** [comment]
Also to what extent do we need UUIDs implemented? Right now I'm assuming they will be char arrays in the packet structs and we just do a memcpy

**Comment by amlweems** [comment]
I agree that we can represent UUIDs as char arrays for serialization and deserialization. I can't think of a case where we would need to operate on the 128-bit integer itself.

**Comment by JDongian** [comment]
I'm +1 with the current `debug` in the Makefile.
+1 on UUIDs as char arrays; they are basically fixed-length strings.

File: src/marshal.c [hunk 3]
```diff
@@ -135,49 +125,22 @@ size_t format_sizeof(char c)
...
+
+    if(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+        memcpy(&value, buf, size);
+    else
+        memcpy(&value + sizeof(value) - size, buf, size);
+
+    return value;
 }
 
 void reentrant_memmove(void *dest, void *src, size_t len)
```
Comment (JDongian) [inline src/marshal.c]: delete this

**Comment by SrsBusiness** [comment]
does hexdump need to go?

**Comment by JDong820** [comment]
@SrsBusiness yes.

**Comment by SrsBusiness** [comment]
good to merge?

File: src/marshal.c [hunk 4]
```diff
@@ -135,83 +125,53 @@ size_t format_sizeof(char c)
...
-        buffer += size;
-        break;
-    }
-    return buffer;
+    memcpy(buffer, data, size);
+    return (char *)buffer + size;
 }
 
 // Read size bytes from buf as an integer and
 // returns the 128 bit integer at the top of the buffer
```
Comment (JDongian) [inline src/marshal.c]: fix the comment?

**Comment by JDongian** [comment]
Looks mostly good, but at least one comment needs to be updated.

**Comment by SrsBusiness** [comment]
Comment fixed. Also the -p flag is useful to know for git rebase.

File: test/packet_test.c [hunk 5]
```diff
@@ -17,6 +17,11 @@ int random_fmt(char *, int);
 
 FILE *urandom;
 
+
+void hexdump(char *title, void *buffer, int len) {
+    return;
```
Comment (JDongian) [inline test/packet_test.c]: uh what

File: src/marshal.h [hunk 6]
```diff
@@ -7,14 +7,14 @@
 #define expect_more(x) (x & 0x80)
 // align pointer x to y
 // y is assumed to be a power of 2
-#define align(x, y) ((size_t)(x + y - 1) & (~y + 1))
+#define align(x, y) ((uint64_t)((x) + (y) - 1) & (uint64_t)(~(y) + 1))
```
Comment (JDongian) [inline src/marshal.h]: I don't get it

File: test/packet_test.c (hunk 5 above)
Comment (SrsBusiness) [inline test/packet_test.c]: I was lazy.

**Pull request merged**